corehq/apps/ota/views.py

```diff
@@ -61,15 +68,23 @@ def restore(request, domain, app_id=None):
     tags = [
         u'status_code:{}'.format(response.status_code),
     ]
+    env = settings.SERVER_ENVIRONMENT
+    if (env, domain) in settings.RESTORE_TIMING_DOMAINS:
+        tags.append(u'domain:{}'.format(domain))
     datadog_counter('commcare.restores.count', tags=tags)
     if timing_context is not None:
+        datadog_histogram(
+            'commcare.restores.total_time',
+            timing_context.duration,
+            tags=tags,
+        )
         for timer in timing_context.to_list(exclude_root=True):
-            # Only record leaf nodes so we can sum to get the total
-            if timer.is_leaf_node:
+            if timer.name in TIMED_RESTORE_SEGMENTS:
+                segment = TIMED_RESTORE_SEGMENTS[timer.name]
                 datadog_histogram(
```

Comment:
> On ICDS we definitely have enough traffic for histogram.

I'm not sure I agree (given a 10s aggregation window). The graph below is a roughly one-day window of ICDS data from the past week, and includes the busiest time of the day. Except under the very highest spikes, look how closely the average and max times track eachother, and I see almost no difference between 95th percentile and max. This tells me our aggregation window is too small, but I'm not sure we can change that without globally affecting other datadog stats? Maybe ICDS sync times are fairly uniform across all devices?

https://app.datadoghq.com/dash/321184/restore-timing?live=false&page=0&is_auto=false&from_ts=1499814729885&to_ts=1499888539721&tile_size=m&tpl_var_env=icds&fullscreen=220044309

Compare that to the last day of latency data coming in from Riak on prod. This shows latency for both GET and PUT requests. In this case the average is a mostly flat line, 95th percentile is a slightly more jagged line above that and max is most jagged of all. Note that this data is not recorded with datadog's histogram; it's generated internally in riak, but I would expect the general trends to be similar.

https://app.datadoghq.com/dash/175518/riak-cs-system-resources?live=true&page=0&is_auto=false&from_ts=1499971874621&to_ts=1500058274621&tile_size=s&tpl_var_hosts=group%3Ariakcs_new&fullscreen=175562724&priority=all&abstraction_level=1&aggregate_up=false&per_page=30&use_date_happened=true&display_timeline=false&tpl_var_env=production